### ISSUE TITLE
Bundler is downloading each gemspec twice

### ISSUE THREAD

**Issue by mperham** [bundler/bundler#6634]
This is an offshoot of #6216.  I noticed that when I `bundle up foobar`, Bundler will download the gemspecs twice.

```
127.0.0.1 - - [22/Jul/2018:08:32:31 PDT] "GET /quick/Marshal.4.8/foobar-1.0.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-1.0.0.gemspec.rz
127.0.0.1 - - [22/Jul/2018:08:32:31 PDT] "GET /quick/Marshal.4.8/foobar-1.1.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-1.1.0.gemspec.rz
127.0.0.1 - - [22/Jul/2018:08:32:31 PDT] "GET /quick/Marshal.4.8/foobar-2.0.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-2.0.0.gemspec.rz
127.0.0.1 - - [22/Jul/2018:08:32:38 PDT] "GET /quick/Marshal.4.8/foobar-2.0.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-2.0.0.gemspec.rz
127.0.0.1 - - [22/Jul/2018:08:32:38 PDT] "GET /quick/Marshal.4.8/foobar-1.1.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-1.1.0.gemspec.rz
127.0.0.1 - - [22/Jul/2018:08:32:38 PDT] "GET /quick/Marshal.4.8/foobar-1.0.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-1.0.0.gemspec.rz
```

At the same time, DEBUG=1 prints this message:

```
Double checking for all specs (due to the size of the request) in rubygems repository
http://localhost:4000/, https://rubygems.org/ or installed locally
```

I've created a repo which reproduces the problem, see https://github.com/mperham/old_source_6216

**Comment by mperham** [comment]
It's also kind of odd that it pauses 7 seconds between the first and second round of downloads.  I can't imagine what would take 7 seconds when we are talking about a Gemfile with one gem in it! (I'm using 1.16.2)

**Comment by segiddins** [comment]
Thanks! I’ll look into this as soon as I get a chance!

**Comment by segiddins** [comment]
When I ran this, I'm not seeing duplicated requests:

```
[2018-07-22 17:29:37] ERROR `/versions' not found.
::1 - - [22/Jul/2018:17:29:37 PDT] "GET /versions HTTP/1.1" 404 278
- -> /versions
[2018-07-22 17:29:37] ERROR `/api/v1/dependencies' not found.
::1 - - [22/Jul/2018:17:29:37 PDT] "GET /api/v1/dependencies HTTP/1.1" 404 289
- -> /api/v1/dependencies
::1 - - [22/Jul/2018:17:29:37 PDT] "GET /specs.4.8.gz HTTP/1.1" 200 100
- -> /specs.4.8.gz
::1 - - [22/Jul/2018:17:29:37 PDT] "GET /prerelease_specs.4.8.gz HTTP/1.1" 200 24
- -> /prerelease_specs.4.8.gz
::1 - - [22/Jul/2018:17:29:44 PDT] "GET /quick/Marshal.4.8/foobar-2.1.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-2.1.0.gemspec.rz
::1 - - [22/Jul/2018:17:29:44 PDT] "GET /quick/Marshal.4.8/foobar-2.0.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-2.0.0.gemspec.rz
::1 - - [22/Jul/2018:17:29:44 PDT] "GET /quick/Marshal.4.8/foobar-1.1.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-1.1.0.gemspec.rz
::1 - - [22/Jul/2018:17:29:44 PDT] "GET /quick/Marshal.4.8/foobar-1.0.0.gemspec.rz HTTP/1.1" 200 337
- -> /quick/Marshal.4.8/foobar-1.0.0.gemspec.rz\
```

**Comment by mperham** [comment]
Are you using 1.16.2 and running “bundle up foobar”?

**Comment by segiddins** [comment]
😳 I totally failed to read that and was just running `bundle install`. Let me try again, my apologies

**Comment by segiddins** [comment]
Finally reproduced this! Looks like it's fixed in "bundler 2 mode", though, so if you just clone master and try that out it should only download each gem spec once!

**Comment by mperham** [comment]
I have thousands of Ruby developers hitting my gem server; I can't upgrade them.  They are running into this bug.  What will be the first version with the fix?

**Comment by segiddins** [comment]
2.0, most likely

**Comment by colby-swandale** [comment]
This issue looks like it has already been fixed. Closing.
